libs/sync/sync.go

```diff
@@ -216,6 +175,40 @@ func (s *Sync) RunOnce(ctx context.Context) error {
 	return nil
 }
 
+func getFileList(ctx context.Context, s *Sync) ([]fileset.File, error) {
+	// tradeoff: doing portable monitoring only due to macOS max descriptor manual ulimit setting requirement
+	// https://github.com/gorakhargosh/watchdog/blob/master/src/watchdog/observers/kqueue.py#L394-L418
+	all := core.NewSetF(func(f fileset.File) string {
+		return f.Absolute
+	})
+	gitFiles, err := s.fileSet.All()
+	if err != nil {
+		log.Errorf(ctx, "cannot list files: %s", err)
+		return nil, err
+	}
+	all.Add(gitFiles...)
```

Comment:
Does this work fine even for very large slices?

Reply:
@pietern it doesn't look like there are any limitations in docs and I also ran this to experiment

```
func pLen(nums ...int) {
	fmt.Println(len(nums))
}

func TestX(t *testing.T) {

	nums := make([]int, (10 * 1000 * 1000 * 1000))
	pLen(nums...)
}

```

So it worked with 10 billion items correctly

Reply:
Thanks for checking. It must be passed by reference already (in C varargs live on the stack, IIRC).